runtime/compiler/runtime/JITClientSession.cpp

```diff
@@ -778,7 +778,31 @@ ClientSessionData::getOrCreateAOTCache(JITServer::ServerStream *stream)
       if (auto aotCacheMap = TR::CompilationInfo::get()->getJITServerAOTCacheMap())
          {
          _aotCache = aotCacheMap->get(_aotCacheName, _clientUID);
-         _aotHeaderRecord = _aotCache->getAOTHeaderRecord(&_vmInfo->_aotHeader, _clientUID);
+         if (!_aotCache)
+            {
+            _vmInfo->_useAOTCache = false;
+            if (TR::Options::getVerboseOption(TR_VerboseJITServer))
+               TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer,
+                  "clientUID=%llu requested AOT cache but the AOT cache size limit has been reached, disabling AOT cache",
+                  (unsigned long long)_clientUID
+               );
+            return NULL;
+            }
+
+         auto record = _aotCache->getAOTHeaderRecord(&_vmInfo->_aotHeader, _clientUID);
+         if (record)
+            {
+            _aotHeaderRecord = record;
+            }
+         else
+            {
+            _vmInfo->_useAOTCache = false;
```

Comment:
Need to set `_aotCache` to `NULL` as well, otherwise this client may still try to use it in some cases.